spec/unit/lib/cloud_controller/permissions_spec.rb

```diff
@@ -429,115 +429,120 @@ module VCAP::CloudController
         it 'returns true for space developer' do
           org.add_user(user)
           space.add_developer(user)
-          expect(permissions.can_read_from_space?(space_guid, org_guid)).to be true
+          expect(permissions.can_read_from_space?(space.id, org.id)).to be true
         end
 
         it 'returns true for space manager' do
           org.add_user(user)
           space.add_manager(user)
-          expect(permissions.can_read_from_space?(space_guid, org_guid)).to be true
+          expect(permissions.can_read_from_space?(space.id, org.id)).to be true
         end
 
         it 'returns true for space auditor' do
           org.add_user(user)
           space.add_auditor(user)
-          expect(permissions.can_read_from_space?(space_guid, org_guid)).to be true
+          expect(permissions.can_read_from_space?(space.id, org.id)).to be true
         end
 
         it 'returns true for org manager' do
           org.add_manager(user)
-          expect(permissions.can_read_from_space?(space_guid, org_guid)).to be true
+          expect(permissions.can_read_from_space?(space.id, org.id)).to be true
         end
       end
     end
 
     describe '#can_read_secrets_in_space?' do
+      let(:membership) { instance_double(Membership) }
+
+      before do
+        allow(Membership).to receive(:new).with(user).and_return(membership)
+      end
       context 'user has no membership' do
+        before do
+          allow(membership).to receive(:has_any_roles?).with(Permissions::ROLES_FOR_SPACE_SECRETS_READING, space.id, org.id).and_return(false)
+        end
         context 'and user is an admin' do
           it 'returns true' do
             set_current_user_as_admin
-            expect(permissions.can_read_secrets_in_space?(space_guid, org_guid)).to be true
+            expect(permissions.can_read_secrets_in_space?(space.id, org.id)).to be true
           end
         end
 
         context 'and user is admin_read_only' do
           it 'returns true' do
             set_current_user_as_admin_read_only
-            expect(permissions.can_read_secrets_in_space?(space_guid, org_guid)).to be true
+            expect(permissions.can_read_secrets_in_space?(space.id, org.id)).to be true
           end
         end
 
         context 'and user is global auditor' do
           it 'return false' do
             set_current_user_as_global_auditor
-            expect(permissions.can_read_secrets_in_space?(space_guid, org_guid)).to be false
+            expect(permissions.can_read_secrets_in_space?(space.id, org.id)).to be false
           end
         end
 
         context 'and user is not an admin' do
           it 'return false' do
             set_current_user(user)
-            expect(permissions.can_read_secrets_in_space?(space_guid, org_guid)).to be false
+            expect(permissions.can_read_secrets_in_space?(space.id, org.id)).to be false
           end
         end
       end
 
       context 'user has valid membership' do
```

Comment:
The previous checks for different roles also worked fine. Any reason why not to keep them?

Reply:
So yeah, as you can see my intention was to mock `Membership` - I felt it that was better than testing the behaviour of both classes in `permissions_spec.rb` - and then stub its various methods to always return the same thing. That meant that it was only necessary to test _one_ of the non-global reader roles, since they'd all just give the same answer.

But if you'd rather I don't mock `Membership` then it makes sense to leave these in.